src/main/java/uk/ac/cam/cl/dtg/segue/api/GroupsFacade.java

```diff
@@ -131,6 +131,90 @@ public Response getGroupsForCurrentUser(@Context final HttpServletRequest reques
         }
     }
 
+    /**
+     * Get all groups where the user is a member.
+     *
+     * @param request            - so we can identify the current user.
+     * @param cacheRequest       - so that we can control caching of this endpoint
+     * @param archivedGroupsOnly - include archived groups in response - default is false - i.e. show only unarchived
+     * @return List of groups for the current user.
+     */
+    @GET
+    @Path("/membership")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getGroupMembership(@Context final HttpServletRequest request,
+                                            @Context final Request cacheRequest, @QueryParam("archived_groups_only") final boolean archivedGroupsOnly) {
+        try {
+            RegisteredUserDTO user = userManager.getCurrentRegisteredUser(request);
+            List<UserGroupDTO> groups = groupManager.getGroupMembershipList(user);
```

Comment:
There is some serious potential for repeated identical queries hidden in here. Not that we can do anything about this in this PR, but we may want to start becoming more aware of these issues.